src/ProjectTemplates/Web.Spa.ProjectTemplates/Angular-CSharp.njsproj.in

```diff
@@ -0,0 +1,163 @@
+<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.0">
+  <PropertyGroup>
+    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">14.0</VisualStudioVersion>
+    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
+    <Name>Company.WebApplication1.Client</Name>
+    <RootNamespace>Company.WebApplication1.Client</RootNamespace>
+    <LaunchUrl>https://localhost:5002</LaunchUrl>
+    <SaveNodeJsSettingsInProjectFile>True</SaveNodeJsSettingsInProjectFile>
+  </PropertyGroup>
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <SchemaVersion>2.0</SchemaVersion>
+    <ProjectGuid>5990939C-7E7B-4CFA-86FF-44CA5756498A</ProjectGuid>
+    <ProjectHome>.</ProjectHome>
+    <StartupFile>node_modules/@angular/cli/bin/ng</StartupFile>
+    <AngularFrontendPort>5002</AngularFrontendPort>
+    <AspNetHttpsCertificate>$(MSBuildProjectDirectory)/obj/EFB8BE63-2B17-4435-B7AD-E9FAA9964C16.crt</AspNetHttpsCertificate>
+    <SearchPath>
+    </SearchPath>
+    <WorkingDirectory>.</WorkingDirectory>
+    <OutputPath>.</OutputPath>
+    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <ProjectTypeGuids>{3AF33F2E-1136-4D97-BBB7-1795711AC8B8};{349c5851-65df-11da-9384-00065b846f21};{9092AA53-FB77-4645-B42D-1CCCA6BD08BD}</ProjectTypeGuids>
+    <NodejsPort>5002</NodejsPort>
+    <StartWebBrowser>True</StartWebBrowser>
+    <ScriptArguments>serve --proxy-config proxy.conf.json --port 5002 --ssl --ssl-key ./obj/65dbd9db-07fc-4667-bfff-eb58cc8cdeb0.key --ssl-cert ./obj/65dbd9db-07fc-4667-bfff-eb58cc8cdeb0.pem</ScriptArguments>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)' == 'Debug' ">
+    <DebugSymbols>true</DebugSymbols>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
+    <DebugSymbols>true</DebugSymbols>
+  </PropertyGroup>
+  <ItemGroup>
+    <Content Include=".editorconfig" />
```

Comment:
So `njsproj` requires every file to be listed explicitly? Does anything go wrong when developers add new files but don't list them here?

My guess is it just means they won't show up in VS. So:

 * If all developers on a team are using VS, then they'd add files using VS, and the project would be updated. Everyone's happy.
 * If all developers on a team are *not* using VS, then they add files some other way and the project isn't updated. But nobody cares because they don't need VS to know about the files, so everyone's happy.
 * If some developers on a team are using VS and others aren't, then everyone gets cross with each other... :)

Maybe that third case is something we can live with. But if it's any more complicated than this, it makes me question whether such a project type is a good idea.

Reply:
There is a fourth scenario though:

 * Developers try to scaffold new files into their project using their SPA framework's CLI too.

Does VS notice this and auto-add the file? Seems quite important to do so.

Reply:
This is a result of the project being and old-style projects. All your comments are correct here. 

I've tested and globs can be used, however their Vue template doesn't use them, so I followed up the pattern there, but it is something on my pending list of things to talk about.